src/main/java/uk/gov/companieshouse/mapper/CompanyClosableMapper.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.companieshouse.mapper;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.model.enums.CompanyType;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Service
+public class CompanyClosableMapper {
```

Comment:
I still don't believe this is a mapper - it's more of a checker. There should be no logic in a mapper, and this class is logic-only

Reply:
Ahh OK. Move it to the validator?

Reply:
Or rename it to validator to keep the logic compartmentalised?

Reply:
a rename should suffice. Make sure to update the package too